### DISCUSSION TITLE
想了解遺下大家對自己科系課程的感想

### DISCUSSION THREAD

**Discussion by lai1008** [1101-GS4719-C/Python-Community-GS4719-C#2223] [General]
想問一夏大家對自己的科系有沒有甚麼課覺得很推薦外系的學生去修或是有沒有甚麼課超硬，硬到會想要停修

**Comment by lai1008** [comment]
太空系的工程實習真的超硬，教很多很難的程式或是一些軟體，對大一的學生來說真的很難

**Reply by zoo100130** [reply]
我每次看到太空系的都由衷感到羨慕，感覺課程充實又好玩，但可能得花很多時間，但覺得能學到真的有用的東西

**Comment by yuan3x2** [comment]
資工系超讚，很多課給分都很甜，再加上能學習到許多應用的技術原理層面，還能鍛鍊自學能力！我個人很推薦陳弘軒的課，教授講課淺顯易懂，用心教學，超讚的。

**Comment by ed112601513** [comment]
大氣系的課我覺得還不錯，特別是王國英教授的課程，他會帶著學生一起到櫻花樹下打程式，上課生動活潑並且常常傳輸給我們python is beautiful的觀念，讓我覺得獲益匪淺。

**Comment by zoo100130** [comment]
機械系:精密機械設計
修到會想休學，硬到會懷疑人生那種，我覺得應該會留下心理陰影，很多東西上網查也不一定能找到答案，大量的trial and error，再花幾十個小時畫圖之後，因為一個小修改整個專案打掉重練是真的哭不來，更刺激的是這堂課需要team work，一個組員爆就連環爆。
但對熱愛的人來說應該是一個練功的好機會，整個專案會學到設計分析(要用手算之後再自己key 程式跑function)、畫圖技巧、專案管理、文件和版本管理的能力都會大幅提升。

**Comment by Boying12** [comment]
作為一名電機系的大四學生，對於我們科系的課程，我有不少感想想與大家分享。在我看來，電機系的課程既有趣又具挑戰性，但有些課程的確是硬到讓人想要停修的。
首先，計算機概論這門課對於任何有興趣進入電機工程領域的學生來說，都是一個很好的起點。這門課雖然內容廣泛，涵蓋了計算機的基本概念，但對於沒有基礎的外系學生來說，可能會有些困難。若對程式設計和計算機結構感興趣的話，這門課還是值得一修的。
接著，電磁學則是讓我感到相當頭痛的一門課。這門課涉及的數學和物理概念非常抽象，對於理解電磁場的運作機制需要相當深厚的基礎。很多同學在學習過程中都會感到迷茫，甚至出現想要停修的念頭，尤其是面對那些繁瑣的公式和計算時。
然後是數位系統設計與實作，這門課雖然在實作上很有趣，但在理論部分的理解上卻讓人感到挑戰。學習邏輯電路、時序電路，還有如何設計一個完整的數位系統，對於一些同學來說，可能會感到壓力山大。如果對數位電路有興趣，這門課還是很值得一修的，因為它對未來的工作是非常有幫助的。
最後，超大型積體電路設計也是一門難度極高的課程，很多人甚至會考慮停修。
總的來說，電機系的課程雖然有些難度很高，但如果你對這些領域有興趣，還是值得去挑戰的。每一門課的學習都能幫助我們建立更堅實的基礎，並且在未來的職業生涯中受益良多。希望能有更多的同學進來體驗這樣的學習旅程！
